src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IDelegateCreationExpression.vb

```diff
@@ -3502,6 +3686,176 @@ BC30469: Reference to a non-shared member requires an object reference.
+
+Public Class C
+    Public Sub M1()
+        CType(((AddressOf M2)), Action)'BIND:"CType(((AddressOf M2)), Action)"
+    End Sub
+
+    Public Sub M2()
+    End Sub
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IDelegateCreationOperation (OperationKind.DelegateCreation, Type: System.Action) (Syntax: 'CType(((Add ... )), Action)')
+  Target: 
+    IParenthesizedOperation (OperationKind.Parenthesized, Type: System.Action) (Syntax: '((AddressOf M2))')
+      Operand: 
+        IParenthesizedOperation (OperationKind.Parenthesized, Type: System.Action) (Syntax: '(AddressOf M2)')
+          Operand: 
+            IMethodReferenceOperation: Sub C.M2() (OperationKind.MethodReference, Type: null) (Syntax: 'AddressOf M2')
+              Instance Receiver: 
+                IInstanceReferenceOperation (OperationKind.InstanceReference, Type: C, IsImplicit) (Syntax: 'M2')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of CTypeExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub DelegateCreation_ParenthesizedAddressOf_CType_InvalidMethod()
+            Dim source = <![CDATA[
+Imports System
+
+Public Class C
+    Public Sub M1()
+        CType(((AddressOf M2)), Action)'BIND:"CType(((AddressOf M2)), Action)"
+    End Sub
+
+    Public Sub M2(o As Object)
+    End Sub
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IDelegateCreationOperation (OperationKind.DelegateCreation, Type: System.Action, IsInvalid) (Syntax: 'CType(((Add ... )), Action)')
+  Target: 
+    IParenthesizedOperation (OperationKind.Parenthesized, Type: System.Action, IsInvalid) (Syntax: '((AddressOf M2))')
+      Operand: 
+        IParenthesizedOperation (OperationKind.Parenthesized, Type: System.Action, IsInvalid) (Syntax: '(AddressOf M2)')
+          Operand: 
+            IOperation:  (OperationKind.None, Type: null, IsInvalid) (Syntax: 'AddressOf M2')
+              Children(1):
+                  IOperation:  (OperationKind.None, Type: null, IsInvalid) (Syntax: 'M2')
+                    Children(1):
+                        IInstanceReferenceOperation (OperationKind.InstanceReference, Type: C, IsInvalid, IsImplicit) (Syntax: 'M2')
+]]>.Value
+
+            Dim expectedDiagnostics = <![CDATA[
+BC31143: Method 'Public Sub M2(o As Object)' does not have a signature compatible with delegate 'Delegate Sub Action()'.
+        CType(((AddressOf M2)), Action)'BIND:"CType(((AddressOf M2)), Action)"
+                          ~~
+]]>.Value
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of CTypeExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub DelegateCreation_ParenthesizedAddressOf_CType_InvalidCast()
```

Comment:
>InvalidCast [](start = 65, length = 11)

Same problem as with a test for lambda

Reply:
Added.

---
In reply to: [152435890](https://github.com/dotnet/roslyn/pull/23315#discussion_r152435890) [](ancestors = 152435890)